spec/index.bs

```diff
@@ -4600,21 +4610,34 @@ interface mixin GPUProgrammablePassEncoder {
             **Called on:** {{GPUProgrammablePassEncoder}} this.
 
             **Arguments:**
-            <!--The overload appears to be confusing bikeshed, and it ends up expecting this to
+            <pre class=argumentdef for="GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)">
+                |index|: The index to set the bind group at.
+                |bindGroup|: Bind group to use for subsequent render or compute commands.
+
+                <!--The overload appears to be confusing bikeshed, and it ends up expecting this to
                 define the arguments for the 5-arg variant of the method, despite the "for"
                 explicitly pointing at the 3-arg variant.-->
-            <!--pre class=argumentdef for="GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)">
-                index:
-                bindGroup:
-                dynamicOffsets:
-            </pre-->
+                <!--|dynamicOffsets|: Array containing buffer offsets in bytes for each entry in
+                    |bindGroup| with marked as {{GPUBindGroupLayoutEntry/hasDynamicOffset}}.-->
+            </pre>
 
             Issue: Resolve bikeshed conflict when using `argumentdef` with
                 {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}}.
 
             **Returns:** void
 
-            Issue: Describe {{GPUProgrammablePassEncoder/setBindGroup(index, bindGroup, dynamicOffsets)}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|.{{GPUObjectBase/[[device]]}}:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, make |this| [=invalid=] and stop.
+                    <div class=validusage>
+                        - |bindGroup| is [=valid=].
+                        - |bindGroup|.{{GPUObjectBase/[[device]]}} is |this|.{{GPUObjectBase/[[device]]}}.
+                        - |index| &lt; {{GPULimits/maxBindGroups|GPULimits.maxBindGroups}}.
+                        - dynamicOffsets.length is
+                            |bindGroup|.{{GPUBindGroup/[[layout]]}}.{{GPUBindGroupLayout/[[dynamicOffsetCount]]}}.
```

Comment:
we also need to check that the `binding offset + dynamic offset + minimumBufferSize is <= buffer size`, for each binding with dynamic offset.

Reply:
Right, forgot that we can validate this here rather than during the draw call.

I took a stab at it but... wow. There's just no clean way to document this validation! Feedback on how to help it read better is very welcome.